plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/internal/MavenDependencyVersionsFetcherGCS.kt

```diff
@@ -1,61 +1,83 @@
+            val storage = StorageOptions.newBuilder()
+                .setCredentials(ServiceAccountCredentials.fromStream(FileInputStream(CREDENTIALS_PATH)))
+                .build()
+                .service
 
-        val allVersions = parseVersionsFromMavenMetaData(xml)
+            bucket = storage.get(bucketName) ?: throw BuildCacheException("$bucketName is unavailable")
 
-        return SuccessfulResult(
-            lastUpdateTimestampMillis = parseLastUpdatedFromMavenMetaData(xml),
-            availableVersions = if (versionFilter == null) {
-                allVersions
-            } else {
-                allVersions.filter(versionFilter)
-            }
-        )
+        } catch (e: FileNotFoundException) {
+            throw GradleException("Unable to load credentials from $CREDENTIALS_PATH.", e)
+        } catch (e: IOException) {
+            throw GradleException("Unable to access Google Cloud Storage bucket '$bucketName'.", e)
+        } catch (e: StorageException) {
+            throw GradleException("Unable to access Google Cloud Storage bucket '$bucketName'.", e)
+        }
     }
 
-    companion object {
-        @Suppress("SpellCheckingInspection")
-        private val mavenLastUpdatedDateFormat = SimpleDateFormat("yyyyMMddHHmmss", Locale.ROOT).apply {
-            timeZone = TimeZone.getTimeZone("UTC")
-        }
+    @OptIn(ExperimentalStdlibApi::class)
+    fun getBlob(path: String): String {
+        val fullPath = listOfNotNull(repoPath, path).joinToString("/", "", "")
+        try {
+            val blob = bucket.get(fullPath) ?: throw FileNotFoundException("blob $fullPath is missing")
+            val stringContent = blob.getContent().decodeToString()
 
-        private fun parseLastUpdatedFromMavenMetaData(xml: String): Long = runCatching {
-            val dateString = xml.substringAfter("<lastUpdated>").substringBefore("</lastUpdated>")
-            return mavenLastUpdatedDateFormat.parse(dateString).time
-        }.getOrDefault(0)
+            System.err.println("returning $stringContent")
+            return stringContent
 
+        } catch (e: StorageException) {
+            // https://github.com/googleapis/google-cloud-java/issues/3402
+            if (e.message?.contains("404") == true) {
+                throw FileNotFoundException("missing blob")
+            }
 
-        private fun parseVersionsFromMavenMetaData(xml: String): List<Version> {
-            return xml.substringAfter("<versions>").substringBefore("</versions>")
-                .split("<version>", "</version>")
-                .mapNotNull { if (it.isBlank()) null else Version(it.trim()) }
+            throw FileNotFoundException("Unable to load '$fullPath' from Google Cloud Storage bucket '$bucketName'.")
         }
+
+    }
+
+    companion object {
+        private val CREDENTIALS_PATH = System.getenv("GOOGLE_APPLICATION_CREDENTIALS")
+            ?: error("required environment variable 'GOOGLE_APPLICATION_CREDENTIALS' missing")
```

Comment:
Is that a standard or something specific to your project's or company's conventions?
If it's standard, please comment with the link here.

Reply:
https://cloud.google.com/docs/authentication/getting-started#setting_the_environment_variable

Reply:
it is the same as what gradle expects there to be